### ISSUE TITLE
Can't send mail

### ISSUE THREAD

**Issue by ikkifujiwara** [directus/api#1205]
# Bug Report

I'm using Directus Suite and i want to create a new action hook to send email on after create / update item. But i keep bumping into this error can't send email (logs below). I'm using GMAIL smtp for mail configuration.

## Other Context & Screenshots

public/extensions/custom/hook/Webhook/hooks.php

```
<?php

return [
    'actions' => [
        'item.create.edm:after' => '\Directus\Custom\Hooks\Webhook\Classes\EdmSendEmail',
        'item.update.edm:after' => '\Directus\Custom\Hooks\Webhook\Classes\EdmSendEmail'
    ]
];
```

public/extensions/custom/hook/Webhook/Classes/EdmSendEmail.php

```
<?php

namespace Directus\Custom\Hooks\Webhook\Classes;

use Directus\Hook\HookInterface;
use Directus\Hook\Payload;

class EdmSendEmail
{
    public function __invoke(array $data)
    {
        $container = \Directus\Application\Application::getInstance()->getContainer();

        try {
            $itemsService = new \Directus\Services\ItemsService($container);
            $edm = $itemsService->find('edm', $data['id'], [
                'fields' => '*,template.*,signature.signature_id.*'
            ]);
            $data = $edm['data'];

            $subject = $data['title'];
            $body = $data['template']['template'];

            $mailService = new \Directus\Services\MailService($container);
            $mailService->send([
                'type' => 'html',
                'subject' => $subject,
                'body' => $body,
                // 'data' => 'array',
                // 'to' => explode(',', $data['email_tester'])
                'to' => 'johndoe@email.com'
            ]);
        } catch (\Exception $e) {
            $logger = $container->get('logger');
            $logger->error($e);
        }
    }
}
```

```
[2019-08-19 15:09:21] api[_].ERROR: Swift_TransportException: Connection could not be established with host localhost [Connection refused #61] in /Users/waimun/Documents/www/directus/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php:277
Stack trace:
#0 /Users/waimun/Documents/www/directus/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php(62): Swift_Transport_StreamBuffer->_establishSocketConnection()
#1 /Users/waimun/Documents/www/directus/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(113): Swift_Transport_StreamBuffer->initialize(Array)
#2 /Users/waimun/Documents/www/directus/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php(79): Swift_Transport_AbstractSmtpTransport->start()
#3 /Users/waimun/Documents/www/directus/src/core/Directus/Mail/Mailer.php(76): Swift_Mailer->send(Object(Directus\Mail\Message))
#4 /Users/waimun/Documents/www/directus/src/helpers/mail.php(62): Directus\Mail\Mailer->sendWithContent('<!DOCTYPE html ...', 'text/html', Object(Closure))
#5 /Users/waimun/Documents/www/directus/src/helpers/mail.php(44): Directus\send_mail_with_content('<!DOCTYPE html ...', 'text/html', Object(Closure))
#6 /Users/waimun/Documents/www/directus/src/core/Directus/Services/MailService.php(42): Directus\send_mail_with_layout('base.twig', 'Organisational ...', Array, 'text/html', Object(Closure))
#7 /Users/waimun/Documents/www/directus/public/extensions/custom/hooks/Webhook/Classes/EdmSendEmail.php(31): Directus\Services\MailService->send(Array)
#8 [internal function]: Directus\Custom\Hooks\Webhook\Classes\EdmSendEmail->__invoke(Array)
#9 /Users/waimun/Documents/www/directus/src/core/Directus/Hook/Emitter.php(291): call_user_func_array(Object(Directus\Custom\Hooks\Webhook\Classes\EdmSendEmail), Array)
#10 /Users/waimun/Documents/www/directus/src/core/Directus/Hook/Emitter.php(119): Directus\Hook\Emitter->executeListeners(Array, Array, 0)
#11 /Users/waimun/Documents/www/directus/src/core/Directus/Hook/Emitter.php(130): Directus\Hook\Emitter->run('item.update.edm...', Array)
#12 /Users/waimun/Documents/www/directus/src/core/Directus/Database/TableGateway/BaseTableGateway.php(1529): Directus\Hook\Emitter->execute('item.update.edm...', Array)
#13 /Users/waimun/Documents/www/directus/src/core/Directus/Database/TableGateway/BaseTableGateway.php(1623): Directus\Database\TableGateway\BaseTableGateway->runHook('item.update.edm...', Array)
#14 /Users/waimun/Documents/www/directus/src/core/Directus/Database/TableGateway/BaseTableGateway.php(878): Directus\Database\TableGateway\BaseTableGateway->runAfterUpdateHooks('edm', Array)
#15 /Users/waimun/Documents/www/directus/vendor/zendframework/zend-db/src/TableGateway/AbstractTableGateway.php(361): Directus\Database\TableGateway\BaseTableGateway->executeUpdate(Object(Zend\Db\Sql\Update))
#16 /Users/waimun/Documents/www/directus/src/core/Directus/Database/TableGateway/BaseTableGateway.php(455): Zend\Db\TableGateway\AbstractTableGateway->updateWith(Object(Zend\Db\Sql\Update))
#17 /Users/waimun/Documents/www/directus/src/core/Directus/Database/TableGateway/RelationalTableGateway.php(498): Directus\Database\TableGateway\BaseTableGateway->updateRecordByArray(Array)
#18 /Users/waimun/Documents/www/directus/src/core/Directus/Services/ItemsService.php(338): Directus\Database\TableGateway\RelationalTableGateway->updateRecord('2', Array, Array)
#19 /Users/waimun/Documents/www/directus/src/endpoints/Items.php(121): Directus\Services\ItemsService->update('edm', '2', Array, Array)
#20 [internal function]: Directus\Api\Routes\Items->update(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Array)
#21 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(41): call_user_func(Array, Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Array)
#22 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/Route.php(356): Slim\Handlers\Strategies\RequestResponse->__invoke(Array, Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Array)
#23 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/AbstractRateLimitMiddleware.php(34): Slim\Route->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#24 [internal function]: Directus\Application\Http\Middleware\AbstractRateLimitMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Slim\Route))
#25 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\UserRateLimitMiddleware), Array)
#26 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Slim\Route))
#27 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Slim\Route))
#28 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/AuthenticationMiddleware.php(105): Slim\Route->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#29 [internal function]: Directus\Application\Http\Middleware\AuthenticationMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#30 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\AuthenticationMiddleware), Array)
#31 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#32 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#33 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/TableGatewayMiddleware.php(25): Slim\Route->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#34 [internal function]: Directus\Application\Http\Middleware\TableGatewayMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#35 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\TableGatewayMiddleware), Array)
#36 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#37 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#38 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\Route->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#39 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/Route.php(334): Slim\Route->callMiddlewareStack(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#40 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/App.php(515): Slim\Route->run(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#41 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/AbstractRateLimitMiddleware.php(34): Slim\App->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#42 [internal function]: Directus\Application\Http\Middleware\AbstractRateLimitMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Directus\Application\Application))
#43 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\IpRateLimitMiddleware), Array)
#44 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Directus\Application\Application))
#45 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Directus\Application\Application))
#46 /Users/waimun/Documents/www/directus/vendor/directus/proxy-detection/src/ProxyDetectionMiddleware.php(30): Slim\App->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#47 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/ProxyMiddleware.php(18): RKA\Middleware\ProxyDetectionMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#48 [internal function]: Directus\Application\Http\Middleware\ProxyMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#49 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\ProxyMiddleware), Array)
#50 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#51 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#52 /Users/waimun/Documents/www/directus/vendor/akrabat/ip-address-middleware/src/IpAddress.php(113): Slim\App->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#53 [internal function]: RKA\Middleware\IpAddress->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#54 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(RKA\Middleware\IpAddress), Array)
#55 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#56 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#57 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/CorsMiddleware.php(66): Slim\App->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#58 [internal function]: Directus\Application\Http\Middleware\CorsMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#59 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\CorsMiddleware), Array)
#60 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#61 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#62 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Http/Middleware/ResponseCacheMiddleware.php(54): Slim\App->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#63 [internal function]: Directus\Application\Http\Middleware\ResponseCacheMiddleware->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#64 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(Directus\Application\Http\Middleware\ResponseCacheMiddleware), Array)
#65 [internal function]: Slim\DeferredCallable->__invoke(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#66 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response), Object(Closure))
#67 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#68 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/App.php(405): Slim\App->callMiddlewareStack(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#69 /Users/waimun/Documents/www/directus/vendor/slim/slim/Slim/App.php(313): Slim\App->process(Object(Directus\Application\Http\Request), Object(Directus\Application\Http\Response))
#70 /Users/waimun/Documents/www/directus/src/core/Directus/Application/Application.php(161): Slim\App->run(false)
#71 /Users/waimun/Documents/www/directus/public/index.php(5): Directus\Application\Application->run()
#72 /Users/waimun/.composer/vendor/laravel/valet/server.php(158): require('/Users/waimun/D...')
#73 {main} [] []
```

**Comment by bjgajjar** [comment]
@ikkifujiwara - Can you please provide the installation method [Docker / Directus Master Branch]  along with the version?

**Comment by ikkifujiwara** [comment]
> @ikkifujiwara - Can you please provide the installation method [Docker / Directus Master Branch] along with the version?

@bjgajjar using Directus Suite master branch and git clone, version 7.9.0

**Comment by verbman** [comment]
Also receiving the same error while using 3rd party smtp service. Same installation procedure and version as @ikkifujiwara (was working previously). Currently failing on reset password api call.

**Comment by bjgajjar** [comment]
Fixed in #1223
